shared/utils/logic.ts

```diff
@@ -0,0 +1,336 @@
+import { AllowMyInfoBase, FieldBase, FieldResponse } from '../types'
+
+import {
+  FormCondition,
+  FormDto,
+  PreventSubmitLogic,
+  FormLogic,
+  LogicType,
+  ShowFieldLogic,
+  LogicConditionState,
+} from '../types/form'
+
+export type PickLogicSubset<T extends FormDto = FormDto> = Pick<
+  T,
+  | '_id'
+  | 'form_logics'
+  | 'form_fields'
+  | 'responseMode'
+  | 'form_logics'
+  | 'form_fields'
+>
+
+export interface ClientField extends AllowMyInfoBase, FieldBase {
```

Comment:
What is this used for? Can we simplify the `LogicFieldOrResponse` type that uses this type to a simpler type using `Pick` just like `PickLogicSubset`?

Reply:
The previous type `LogicFieldSchemaOrResponse`, was dependent on the type `IClientFieldSchema`

I created this type to replace the `IClientFieldSchema` which had a fieldValue property defined there and which functions like `getCurrentValue` are dependent on. I think `_id` could be retrieved using Pick from FormDto but not so sure how else to add the fieldValue property to this new shared type.

Reply:
Yea, don't worry about it. Probably out of scope for this PR!